goc/bin/copy_config_osg

```diff
@@ -18,21 +18,96 @@ if [ ! -f /etc/cvmfs/repositories.d/config-osg.opensciencegrid.org/client.conf ]
     exit 1
 fi
 
-set -ex
-mkdir -p /cvmfs/config-osg-itb.opensciencegrid.org /etc/cvmfs/config.d
-cat >/etc/cvmfs/config.d/config-osg.opensciencegrid.org.local <<!EOF!
-CVMFS_SERVER_URL=http://oasis-itb.opensciencegrid.org:8000/cvmfs/@fqrn@
-CVMFS_PUBLIC_KEY=/etc/cvmfs/keys/opensciencegrid.org.pub:/srv/etc/cvmfs/oasis-itb.opensciencegrid.org.pub
-CVMFS_HTTP_PROXY=DIRECT
-unset CVMFS_CONFIG_REPOSITORY
-!EOF!
-mount -t cvmfs config-osg.opensciencegrid.org /cvmfs/config-osg-itb.opensciencegrid.org
-if ! cvmfs_server transaction config-osg.opensciencegrid.org; then
-    cvmfs_server abort -f config-osg.opensciencegrid.org
-    cvmfs_server transaction config-osg.opensciencegrid.org
+get_branch()
+{
+    # get branch name
+    raw_current_branch=`git branch | grep "*"`
+    current_branch=${raw_current_branch#* }
+    # get last component of branch name
+    current_branch_name=${current_branch##*\/}
+}
+
+confirm() 
+{
+    # call with a prompt string or use a default
+    read -r -p "${1:-Are you sure? [y/N]} " response
+    case "$response" in
+        [yY][eE][sS]|[yY]) 
+            false
+            ;;
+        *)
+            true
+            ;;
+    esac
+}
+
+# run only on -itb host 
+if [[ $(hostname -s) = *-itb ]]; then
+    # run only if non-root
+    if ! [ $(id -u) = 0 ]; then
+        if git remote -v 2>/dev/null | grep -q /config-repo.git; then
+            current_dir="$(git rev-parse --show-toplevel)"
+            set -ex
+            cd ${current_dir}
+            git pull upstream osg
+            rsync -av /cvmfs/config-osg.opensciencegrid.org ${current_dir}
+        else
+            >&2 echo "Make sure to git clone config-repo.git and you're inside the repo directory, exiting..."
+            exit 1
+        fi
+    else
+        >&2 echo "Make sure to run ${ME} as non-root, exiting..."
+        exit 1
+    fi
+
+else
+
+    if ! [ $(id -u) = 0 ]; then
+       echo "I am not root!"
+       exit 1
+    fi
+
+    set -ex
+    # exit if /opt/config-repo/ doesn't exist
```

Comment:
I'm sorry, I should have said to put this just before the set -ex instead of just after it.  It's very minor, but might as well do it the better way while we're in here and there's still something more substantial left to change.

Reply:
OK.